File: solutions/csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int offset = logLine.IndexOf("]:") + "]:".Length;
        return logLine.Substring(offset).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int start = logLine.IndexOf("[") + 1;
        int end = logLine.IndexOf("]") - 1;
        return logLine.Substring(start, end).ToLower();
    }

    public static string Reformat(string logLine) =>
        $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: solutions/fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string) : string =
    let marker: string = "]:"
    let offset: int = logLine.IndexOf(marker) + marker.Length
    logLine.Substring(offset).Trim()

let logLevel (logLine: string) : string =
    let start: int = logLine.IndexOf("[") + 1
    let length: int = logLine.IndexOf("]") - start
    logLine.Substring(start, length).ToLower()

let reformat (logLine: string) : string =
    $"{message logLine} ({logLevel logLine})"
